seaborn/matrix.py

```diff
@@ -194,11 +194,19 @@ def __init__(self, data, vmin, vmax, cmap, center, robust, annot, fmt,
     def _determine_cmap_params(self, plot_data, vmin, vmax,
                                cmap, center, robust):
         """Use some heuristics to set good defaults for colorbar and range."""
-        calc_data = plot_data.data[~np.isnan(plot_data.data)]
+
+        # plot_data is a np.ma.array instance
+        calc_data = plot_data.filled(np.nan)
         if vmin is None:
-            vmin = np.percentile(calc_data, 2) if robust else calc_data.min()
+            vmin = (
```

Comment:
Given that a normal if/else block will now take up exactly the same number of lines and be more clear, I'd use that syntax here.

Reply:
Done, thanks!